mne/io/nedf/nedf.py

```diff
@@ -0,0 +1,193 @@
+    if headerend == -1:
+        raise RuntimeError('End of header null not found')
+    headerxml = ElementTree.fromstring(header[:headerend])
+    nedfversion = headerxml.findtext('NEDFversion', '')
+    if nedfversion not in ['1.3', '1.4']:
+        warn('Unexpected NEDFversion, hope this works anyway')
+
+    if headerxml.findtext('AdditionalChannelStatus', 'OFF') != 'OFF':
+        raise RuntimeError('Unknown additional channel, aborting.')
+
+    n_acc = int(headerxml.findtext('NumberOfChannelsOfAccelerometer', 0))
+    if n_acc:
+        # expect one sample of u16 accelerometer data per block
+        dt.append(('acc', '>u2', (n_acc,)))
+
+    eegset = headerxml.find('EEGSettings')
+    if eegset is None:
+        raise RuntimeError('No EEG channels found')
+    nchantotal = int(_getsubnodetext(eegset, 'TotalNumberOfChannels'))
+    info['nchan'] = nchantotal
+
+    info['sfreq'] = int(_getsubnodetext(eegset, 'EEGSamplingRate'))
+    info['ch_names'] = [e.text for e in eegset.find('EEGMontage')]
+    # expect nchantotal uint24s
+    datadt.append(('eeg', 'B', (nchantotal, 3)))
+
+    if headerxml.find('STIMSettings') is not None:
+        # 2* -> two stim samples per eeg sample
+        datadt.append(('stim', 'B', (2, nchantotal, 3)))
+
+    if headerxml.findtext('stepDetails/DeviceClass', '') == 'STARSTIM':
+        warn('Found Starstim, this hasn\'t been tested extensively!')
+
+    # Trigger data: 4 bytes in newer versions, 1 byte in older versions
+    trigger_type = '>i4' if headerxml.findtext('NEDFversion') else 'B'
+    datadt.append(('trig', trigger_type))
+    # 5 data samples per block
+    dt.append(('data', np.dtype(datadt), (5,)))
+
+    date = headerxml.findtext('StepDetails/StartDate_firstEEGTimestamp', 0)
+    info['meas_date'] = datetime.fromtimestamp(int(date) / 1000, timezone.utc)
+    return info, np.dtype(dt)
+
+
+def _read_nedf_eeg(filename: str):
+    """Read header info and EEG data from an .nedf file.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the .nedf file.
+
+    Returns
+    -------
+    eeg : array, shape (n_samples, n_channels)
+        Unscaled EEG data.
+    info : dict
+        Information from the file header.
+    triggers : array, shape (n_annots, 2)
+        Start samples and values of each trigger.
+    scale : float
+        Scaling factor for the EEG data.
+    """
+    # the first 10240 bytes are header in XML format, padded with NULL bytes
+    hdrlen = 10240
+    with open(filename, mode='rb') as f:
+        info, dt = _parse_nedf_header(f.read(hdrlen))
+        data = np.fromfile(f, dtype=dt)
+
+    # convert uint8-triplet -> float32
+    eeg = data['data']['eeg'] @ [1 << 16, 1 << 8, 1.]
+    eeg = eeg.reshape((-1, info['nchan']))
+    # convert sign if necessary
+    eeg[eeg > (1 << 23)] -= 1 << 24
+
+    triggers = data['data']['trig'].flatten()
+    triggerind = triggers.nonzero()[0]
+    triggers = np.stack((triggerind, triggers[triggerind])).T
```

Comment:
This I'm not so sure about. In most systems that have digitally sampled TTL trigger channels (which this is, right?) we expose these as `STIM` channels, as this allows using `find_events`. I think it makes sense to continue this practice, rather than convert these to annotations directly.

Reply:
Although a TTL adapter exists, I don't know anyone using it or if it's recorded here (or even if it can encode more than one bit of information). From my understanding, the recording software subscribes to a specific LSL outlet and writes the LSL events to the trigger channel with a duration of a single sample.

Reply:
I see. I'm still somewhat inclined to have this as a STIM channel and have people use `find_events` with it, as it's basically a "channel" sampled at the rate of the data (even if the LSL stream does not actually use TTL pulses).

@agramfort any opinion?